bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Porting/ImportCommand.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Bit.Core.SecretsManager.Commands.Porting;
+using Bit.Core.SecretsManager.Commands.Porting.Interfaces;
+using Bit.Core.SecretsManager.Entities;
+using Bit.Core.SecretsManager.Repositories;
+
+namespace Bit.Commercial.Core.SecretsManager.Commands.Porting;
+
+public class ImportCommand : IImportCommand
+{
+    private readonly IProjectRepository _projectRepository;
+    private readonly ISecretRepository _secretRepository;
+
+    public ImportCommand(IProjectRepository projectRepository, ISecretRepository secretRepository)
+    {
+        _projectRepository = projectRepository;
+        _secretRepository = secretRepository;
+    }
+
+    public async Task<SMImport> ImportAsync(Guid organizationId, SMImport import)
+    {
+        var importedProjects = new List<Project>();
+        var importedSecrets = new List<Secret>();
+
+        try
+        {
+            import = AssignNewIds(import);
+
+            if (import.Projects != null && import.Projects.Any())
+            {
+                importedProjects = (await _projectRepository.ImportAsync(import.Projects.Select(p => new Project
```

Comment:
Should be enough to track the IDs so we can delete them later? Or did you plan on returning the response to the client?

Reply:
I don't think we need to return anything to the client, so I updated that, good point. I also changed it to only capture the id's, and then later, delete them in case of an error.